TOR auto-publish-enabled property not honored

**Expected behavior**

Setting `org.tbk.tor.auto-publish-enabled=false` should disable auto publishing.

**Actual behavior**

Setting `org.tbk.tor.auto-publish-enabled=false` has no effect.
Application fails to configure.

**Steps to reproduce the problem**
  1. Set property `org.tbk.tor.auto-publish-enabled=false`
  2. Run Application
  3. Fail

**Specifications**
  - Version: 0.3.0
  - Platform: MacOS
  - Module: spring-tor-autoconfigure

`implementation("io.github.theborakompanioni:spring-tor-starter:0.3.0")`

Kotlin 1.6.21
Spring Boot 2.7.4
Gradle 7.5


**Additional context**

Log:
```
Caused by: java.lang.IllegalStateException: Cannot publish hidden service for application. Please specify 'server.port' or disable auto publishing with 'org.tbk.tor.auto-publish-enabled=false'
	at org.tbk.tor.spring.config.TorHiddenServiceAutoConfiguration.applicationHiddenServiceDefinition(TorHiddenServiceAutoConfiguration.java:42) ~[spring-tor-autoconfigure-0.3.0.jar:0.3.0]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.3.23.jar:5.3.23]
	... 84 common frames omitted
```

Debug autoconfiguration output:
```
   TorHiddenServiceAutoConfiguration#applicationHiddenServiceDefinition matched:
      - @ConditionalOnBean (types: org.springframework.boot.autoconfigure.web.ServerProperties; SearchStrategy: all) found bean 'server-org.springframework.boot.autoconfigure.web.ServerProperties'; @ConditionalOnMissingBean (names: applicationHiddenServiceDefinition; SearchStrategy: all) did not find any beans (OnBeanCondition)
```
It looks like `@Conditional(OnAutoPublishEnabledAndServerPortSpecified.class)` is not evaluated at all.

--- a/changelog.md
+++ b/changelog.md
@@ -5,6 +5,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+### Fixed
+- fix: tor auto-publish can be disabled (`org.tbk.tor.auto-publish-enabled`)
+
 ### Added
 - disable p2p network activity by default for bitcoin testcontainer (`networkactive=0`)
 

--- a/spring-tor/spring-tor-autoconfigure/src/main/java/org/tbk/tor/spring/config/TorHiddenServiceAutoConfiguration.java
+++ b/spring-tor/spring-tor-autoconfigure/src/main/java/org/tbk/tor/spring/config/TorHiddenServiceAutoConfiguration.java
@@ -58,7 +58,7 @@ public class TorHiddenServiceAutoConfiguration {
     static class OnAutoPublishEnabledAndServerPortSpecified extends AllNestedConditions {
 
         OnAutoPublishEnabledAndServerPortSpecified() {
-            super(ConfigurationPhase.PARSE_CONFIGURATION);
+            super(ConfigurationPhase.REGISTER_BEAN);
         }
 
         @ConditionalOnProperty(name = "org.tbk.tor.auto-publish-enabled", havingValue = "true", matchIfMissing = true)
@@ -121,7 +121,7 @@ public class TorHiddenServiceAutoConfiguration {
         static class OnAutoPublishEnabledAndManagementServerPortSpecified extends AllNestedConditions {
 
             OnAutoPublishEnabledAndManagementServerPortSpecified() {
-                super(ConfigurationPhase.PARSE_CONFIGURATION);
+                super(ConfigurationPhase.REGISTER_BEAN);
             }
 
             @ConditionalOnProperty(name = "org.tbk.tor.auto-publish-enabled", havingValue = "true", matchIfMissing = true)

--- a/spring-tor/spring-tor-autoconfigure/src/test/java/org/tbk/tor/spring/config/TorAutoConfigurationTest.java
+++ b/spring-tor/spring-tor-autoconfigure/src/test/java/org/tbk/tor/spring/config/TorAutoConfigurationTest.java
@@ -111,6 +111,7 @@ public class TorAutoConfigurationTest {
     @Test
     public void torBeansCreatedInWebContext() {
         this.webContextRunner.withUserConfiguration(autoConfigClasses)
+                .withPropertyValues("server.port=13337")
                 .withBean(ServerProperties.class, () -> {
                     // fake a running webserver for the hidden service to bind to
                     ServerProperties serverProperties = new ServerProperties();
@@ -137,6 +138,13 @@ public class TorAutoConfigurationTest {
     public void torBeansCreatedInWebContextWithAutoPublishDisabled() {
         this.webContextRunner.withUserConfiguration(autoConfigClasses)
                 .withPropertyValues("org.tbk.tor.auto-publish-enabled=false")
+                .withPropertyValues("server.port=13337")
+                .withBean(ServerProperties.class, () -> {
+                    // fake a running webserver
+                    ServerProperties serverProperties = new ServerProperties();
+                    serverProperties.setPort(13337);
+                    return serverProperties;
+                })
                 .run(context -> {
                     autoPublishEnabledAndWebAppBeanNames.forEach(name -> {
                         boolean beanWithNameIsAvailable = context.containsBean(name);

